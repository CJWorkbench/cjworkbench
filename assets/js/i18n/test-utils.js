import { I18nProvider } from '@lingui/react'
import React from 'react'
import { shape, object } from 'prop-types'
import { mount, shallow } from 'enzyme'

/*
 * Below lie utils for testing with i18n.
 * See https://lingui.js.org/guides/testing.html
 */
// Create the I18nProvider to retrieve context for wrapping around.
const intlProvider = new I18nProvider({
  language: 'en',
  catalogs: {
    en: {}
  }
}, {})

const {
  linguiPublisher: {
    i18n: originalI18n
  }
} = intlProvider.getChildContext()

// You customize the i18n object here:
export const i18n = {
  ...originalI18n,
  // provide _ macro, for trying to unwrap default/id or just passing down the key
  _: key => {
    if (typeof key.defaults === 'string') return key.defaults
    if (typeof key.id === 'string') return key.id
    return key
  }
}

/**
 * When using Lingui `withI18n` on components, props.i18n is required.
 */
function nodeWithI18nProp (node) {
  return React.cloneElement(node, { i18n })
}

/*
 * Methods to use
 */
export function shallowWithI18n (node, { context } = {}) {
  return shallow(
    nodeWithI18nProp(node),
    {
      context: Object.assign({}, context, { i18n })
    }
  )
}

export function mountWithI18n (node, { context, childContextTypes } = {}) {
  const newContext = Object.assign({}, context, { linguiPublisher: { i18n } })
  /*
   * I18nProvider sets the linguiPublisher in the context for withI18n to get
   * the i18n object from.
   */
  const newChildContextTypes = Object.assign({},
    {
      linguiPublisher: shape({
        i18n: object.isRequired
      }).isRequired
    },
    childContextTypes
  )
  return mount(
    nodeWithI18nProp(node),
    {
      context: newContext,
      childContextTypes: newChildContextTypes
    }
  )
}
