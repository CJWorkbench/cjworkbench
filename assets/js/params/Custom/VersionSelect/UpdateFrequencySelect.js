import React from 'react'
import PropTypes from 'prop-types'
import UpdateFrequencySelectModal from './UpdateFrequencySelectModal'
import { timeDifference } from '../../../utils'
import { trySetWfModuleAutofetchAction, setWfModuleNotificationsAction } from '../../../workflow-reducer'
import { connect } from 'react-redux'
import { Trans, t } from '@lingui/macro'
import { withI18n } from '@lingui/react'

export const UpdateFrequencySelect = withI18n()(class UpdateFrequencySelect extends React.PureComponent {
  static propTypes = {
    i18n: PropTypes.shape({
      // i18n object injected by LinguiJS withI18n()
      _: PropTypes.func.isRequired
    }),
    workflowId: PropTypes.number.isRequired,
    wfModuleId: PropTypes.number.isRequired,
    isAnonymous: PropTypes.bool.isRequired,
    isReadOnly: PropTypes.bool.isRequired,
    lastCheckDate: PropTypes.instanceOf(Date), // null if never updated
    isAutofetch: PropTypes.bool.isRequired,
    fetchInterval: PropTypes.number.isRequired,
    isEmailUpdates: PropTypes.bool.isRequired,
    setEmailUpdates: PropTypes.func.isRequired, // func(wfModuleId, isEmailUpdates) => undefined
    trySetAutofetch: PropTypes.func.isRequired // func(wfModuleId, isAutofetch, fetchInterval) => Promise[response]
  }

  state = {
    isModalOpen: false,
    quotaExceeded: null // JSON response -- contains autofetch info iff we exceeded quota
  }

  handleClickOpenModal = (ev) => {
    if (ev && ev.preventDefault) ev.preventDefault() // <a> => do not change URL
    if (this.props.isReadOnly) return
    if (this.props.isAnonymous) return

    this.setState({
      isModalOpen: true
    })
  }

  handleCloseModal = () => {
    this.setState({
      isModalOpen: false
    })
  }

  setEmailUpdates = (isEmailUpdates) => {
    const { setEmailUpdates, wfModuleId } = this.props
    setEmailUpdates(wfModuleId, isEmailUpdates)
  }

  trySetAutofetch = (isAutofetch, fetchInterval) => {
    const { trySetAutofetch, wfModuleId } = this.props
    return trySetAutofetch(wfModuleId, isAutofetch, fetchInterval)
  }

  render () {
    const { i18n, lastCheckDate, isAutofetch, fetchInterval, isEmailUpdates, workflowId, wfModuleId } = this.props
    const { isModalOpen } = this.state

    return (
      <div className='update-frequency-select'>
        <div className='update-option'>
          <span className='version-box-option'><Trans id='js.params.Custom.VersionSelect.UpdateFrequencySelect.update'>Update</Trans> </span>
          <a
            href='#'
            title={i18n._(t('js.params.Custom.VersionSelect.UpdateFrequencySelect.changeUpdateSettings.hoverText')`change auto-update settings`)}
            className='content-1 ml-1 action-link'
            onClick={this.handleClickOpenModal}
          >
            {isAutofetch ? <Trans id='js.params.Custom.VersionSelect.UpdateFrequencySelect.auto'>Auto</Trans> : <Trans id='js.params.Custom.VersionSelect.UpdateFrequencySelect.manual'>Manual</Trans>}
          </a>
        </div>
        {lastCheckDate ? (
          <div className='last-checked'>
            <Trans id='js.params.Custom.VersionSelect.UpdateFrequencySelect.lastChecked' description="The parameter is a time difference (i.e. something like '4h ago'. The tag is a <time> tag.">
                Checked <time dateTime={this.props.lastCheckDate.toISOString()}>{timeDifference(lastCheckDate, Date.now(), i18n)}</time>
            </Trans>
          </div>
        ) : null}
        {isModalOpen ? (
          <UpdateFrequencySelectModal
            workflowId={workflowId}
            wfModuleId={wfModuleId}
            isEmailUpdates={isEmailUpdates}
            isAutofetch={isAutofetch}
            fetchInterval={fetchInterval}
            setEmailUpdates={this.setEmailUpdates}
            trySetAutofetch={this.trySetAutofetch}
            onClose={this.handleCloseModal}
          />
        ) : null}
      </div>
    )
  }
})

const mapStateToProps = (state, ownProps) => {
  const workflow = state.workflow || {}
  const wfModule = state.wfModules[String(ownProps.wfModuleId)] || {}
  // We need a "default" value for everything: wfModule might be a placeholder

  const lastCheckString = wfModule.last_update_check // JSON has no date -- that's a STring
  const lastCheckDate = lastCheckString ? new Date(Date.parse(lastCheckString)) : null

  return {
    lastCheckDate,
    workflowId: workflow.id,
    isReadOnly: workflow.read_only,
    isAnonymous: workflow.is_anonymous,
    isEmailUpdates: wfModule.notifications || false,
    isAutofetch: wfModule.auto_update_data || false,
    fetchInterval: wfModule.update_interval || 86400
  }
}

const mapDispatchToProps = {
  trySetAutofetch: trySetWfModuleAutofetchAction,
  setEmailUpdates: setWfModuleNotificationsAction
}

export default connect(mapStateToProps, mapDispatchToProps)(UpdateFrequencySelect)
