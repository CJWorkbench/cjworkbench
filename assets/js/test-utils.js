/* globals jest */
import { createStore, applyMiddleware } from 'redux'
import thunk from 'redux-thunk'
import promiseMiddleware from 'redux-promise-middleware'
import errorMiddleware from './error-middleware'
import { workflowReducer } from './workflow-reducer'

// Returns new mock function that returns given json. Used for mocking "get" API calls
export function jsonResponseMock (json) {
  return jest.fn().mockImplementation(() =>
    Promise.resolve(json)
  )
}

// Returns new mock function that gives an OK HTTP response. Use for mocking "set" API calls
export function okResponseMock () {
  return jsonResponseMock(null)
}

// Helper function to return a promise that resolves after all other promise mocks,
// even if they are chained like Promise.resolve().then(...).then(...)
// Technically: this is designed to resolve on the next macrotask
// https://stackoverflow.com/questions/25915634/difference-between-microtask-and-macrotask-within-an-event-loop-context
export function tick () {
  return new Promise(resolve => {
    setTimeout(resolve, 0)
  })
}

export function sleep (ms) {
  return new Promise(resolve => {
    setTimeout(resolve, ms)
  })
}

/**
 * Return `[ notify, promise ]`; `promise` completes when `notify` is called.
 *
 * Usage:
 *
 *     const [ notify, condition ] = createConditionVariable
 *     const done = condition.then(do_something)
 *     await Promise.all([sleep(0.1).then(notify), done])
 */
export function createConditionVariable () {
  let notify
  const promise = new Promise(resolve => { notify = resolve })
  return [notify, promise]
}

/**
 * Mock Redux store with in-production reducers.
 *
 * This is for integration-style tests: they test the reducer _and_ the caller
 * at the same time.
 *
 * Usage:
 *
 *     const api = { undo: jest.fn() }
 *     const store = mockStoreWithReducer({'workflow': ..., ...}, api)
 *     store.dispatch(Actions.undo())
 *     expect(api.undo).toHaveBeenCalled()
 *     expect(store.getState().workflow).toEqual(...)
 */
export function mockStore (initialState, api = {}) {
  const middlewares = [errorMiddleware(), promiseMiddleware, thunk.withExtraArgument(api)]
  const store = createStore(workflowReducer, initialState, applyMiddleware(...middlewares))
  return store
}
