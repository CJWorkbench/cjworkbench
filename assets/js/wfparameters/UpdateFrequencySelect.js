import React from 'react'
import PropTypes from 'prop-types'
import { Modal, ModalHeader, ModalBody, ModalFooter } from 'reactstrap'
import { Form, FormGroup, Label, Input } from 'reactstrap'
import { timeDifference } from '../utils'
import { store, updateWfModuleAction } from '../workflow-reducer'

export default class UpdateFrequencySelect extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      modalOpen: false,
      message: "",
      liveSettings: {
        manual: !this.props.updateSettings.autoUpdateData,
        period: this.props.updateSettings.updateInterval,
        unit: this.props.updateSettings.updateUnits
      }
    };
    this.state.dialogSettings = Object.assign({}, this.state.liveSettings);

    // Allow props to specify a conversion from browser time to displayed time, so tests can run in UTC (not test machine tz)
    if (props.timezoneOffset != undefined) {
      this.state.timezoneOffset = props.timezoneOffset;
    } else {
      this.state.timezoneOffset = 0; // display in browser local time
    }

    this.toggleModal = this.toggleModal.bind(this);
    this.toggleManual = this.toggleManual.bind(this);
    this.updatePeriod = this.updatePeriod.bind(this);
    this.updateUnit = this.updateUnit.bind(this);
    this.saveSettings = this.saveSettings.bind(this);
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.notifications && (!this.props.notifications && this.state.liveSettings.manual)) {
      this.toggleModal();
      this.toggleManual();
      this.setState({
        message: "Choose a frequency to check on new data and click 'Apply'."
      })
    }
  }

  toggleModal() {
    if (!this.props.isReadOnly) {
      this.setState({
        modalOpen: !this.state.modalOpen,
        dialogSettings: Object.assign({}, this.state.liveSettings)
      }, () => {
        if (!this.state.modalOpen && this.state.liveSettings.manual && this.props.notifications) {
          store.dispatch(updateWfModuleAction(this.props.wfModuleId, {
            notifications: false
          }));
        }
      });
    }
  }

  toggleManual() {
    this.setState(
      {dialogSettings: {
        manual: !this.state.dialogSettings.manual,
        period: this.state.dialogSettings.period,
        unit: this.state.dialogSettings.unit
    }});
  }

  updatePeriod(event) {
    this.setState(
        {dialogSettings: {
          manual: this.state.dialogSettings.manual,
          period: event.target.value,
          unit: this.state.dialogSettings.unit
    }});
  }

  updateUnit(event) {
    this.setState(
      {dialogSettings: {
        manual: this.state.dialogSettings.manual,
        period: this.state.dialogSettings.period,
        unit: event.target.value
    }});
  }

  saveSettings() {
    var params = {
      auto_update_data: !this.state.dialogSettings.manual,
      update_interval: this.state.dialogSettings.period,
      update_units: this.state.dialogSettings.unit
    };
    this.props.api.setWfModuleUpdateSettings(this.props.wfModuleId, params);
    this.setState({liveSettings: Object.assign({}, this.state.dialogSettings)}, () => {
      this.toggleModal();
    });
  }


  render() {

    // button highlights
    var highlightManual = 'action-button manual-button ' + (this.state.dialogSettings.manual ? 'button-blue--fill' : 'button-gray ');
    var highlightAuto = 'action-button auto-button ' + (!this.state.dialogSettings.manual ? 'button-blue--fill' : 'button-gray');

    // info shown on Wf Module card
    var manual = this.state.liveSettings.manual;
    var period = this.state.liveSettings.period;
    var unit = this.state.liveSettings.unit;
    var settingsInfo = manual ? 'manual' :'auto'

    var lastChecked = null;
    var now = new Date();
    if (this.props.updateSettings.lastUpdateCheck)
      lastChecked = <div className='content-4 t-m-gray'>
                      Checked {timeDifference(this.props.updateSettings.lastUpdateCheck, now)}
                    </div>

    return (
      <div className='frequency-item'>
        <div>
          <span className='content-3 t-d-gray'>Update </span>
          <span className='content-3 ml-1 t-f-blue test-modal-button' onClick={this.toggleModal}>{settingsInfo}</span>
        </div>
        {lastChecked}
        <Modal isOpen={this.state.modalOpen} toggle={this.toggleModal} className='modal-dialog'>
          <ModalHeader toggle={this.toggleModal} className='dialog-header'>
            <span className='title-4 t-d-gray'>SYNC SETTINGS</span>
          </ModalHeader>
          <ModalBody className='dialog-body'>
            {this.state.message.length > 0 &&
              <div className='info-3 mt-2 mb-5'>{this.state.message}</div>
            }
            <FormGroup>
              <div className="row">
                <div className="col-sm-3">
                  <div onClick={this.toggleManual} className={highlightAuto}>On</div>
                </div>

                <div className="col-sm-9">
                  <div className='info-2'>Automatically save the current version of the workflow and update it with the newest data.</div>
                  <Label for="updateFreq" className='content-3 t-d-gray mt-4 mb-2'>Check for update every</Label>
                  <div className='update-freq-settings update-freq-test-class mb-5'>
                    <Input
                      type="number"
                      onChange={this.updatePeriod}
                      value={this.state.dialogSettings.period}
                      min='1'
                      max='500'
                      name="updateFreq"
                      id="updateFreqNum"
                      className='number-field t-d-gray content-2'>
                    </Input>
                    <Input
                      type="select"
                      value={this.state.dialogSettings.unit}
                      onChange={this.updateUnit}
                      name="updateFreq"
                      id="updateFreqUnit"
                      className='ml-3 input-dropdown'
                    >
                      <option>minutes</option>
                      <option>hours</option>
                      <option id='days-option'>days</option>
                      <option>weeks</option>
                    </Input>
                  </div>
                </div>
              </div>

              <div className="row d-flex align-items-center">
                <div className="col-sm-3">
                  <div onClick={this.toggleManual} className={highlightManual}>Off</div>
                </div>

                <div className="col-sm-9">
                  <div className='info-2'>Check for new data manually.</div>
                </div>
              </div>
            </FormGroup>
          </ModalBody>
          <ModalFooter className='dialog-footer'>
            <div className='action-button button-gray test-cancel-button mr-4' onClick={this.toggleModal}>Cancel</div>
            <div className='action-button button-blue test-ok-button' onClick={this.saveSettings}>Apply</div>

          </ModalFooter>
        </Modal>
      </div>
    );
  }

}

UpdateFrequencySelect.propTypes = {
  api:              PropTypes.object.isRequired,
  updateSettings:   PropTypes.object.isRequired,
  wfModuleId:       PropTypes.number.isRequired
};
