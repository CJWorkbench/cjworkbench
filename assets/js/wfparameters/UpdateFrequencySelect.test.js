import React from 'react'
import UpdateFrequencySelect  from './UpdateFrequencySelect'
import { shallow } from 'enzyme'
import { okResponseMock } from '../test-utils'


describe('UpdateFrequencySelect', () => {

  var today = new Date();
  var day_before = today.setDate(today.getDate() - 2);
  const updateSettings = {
    autoUpdateData: false, // start in Manual mode
    updateInterval: 5,
    updateUnits: 'minutes',
    lastUpdateCheck: day_before
  };
  const api = {
    setWfModuleUpdateSettings: okResponseMock()
  };
  var wrapper;
  var freqNum;
  var freqUnit;
  var modalLink;
  var autoButton;

  beforeEach(() => wrapper = shallow(
    <UpdateFrequencySelect
      isReadOnly={false}
      updateSettings={updateSettings}
      wfModuleId={212}
      api={api}
    />));
  beforeEach(() => freqUnit = wrapper.find('#updateFreqUnit'));
  beforeEach(() => freqNum = wrapper.find('#updateFreqNum'));
  beforeEach(() => modalLink = wrapper.find('.test-modal-button'));
  beforeEach(() => autoButton = wrapper.find('.auto-button'));


  it('Renders correctly when in Private mode and changes are confirmed with OK', (done) => {

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.state().liveSettings.manual).toBe(true);
    expect(freqNum.get(0).props.value).toBe(5);
    expect(wrapper.state().dialogSettings.period).toBe(5);
    expect(freqUnit.get(0).props.value).toBe('minutes');
    expect(wrapper.state().dialogSettings.unit).toBe('minutes');

    expect(modalLink).toHaveLength(1);
    modalLink.first().simulate('click');

    // Dialog should be open
    expect(wrapper).toMatchSnapshot();

    expect(autoButton).toHaveLength(1);
    autoButton.first().simulate('click');

    freqNum.simulate('change', {target: {value: 888}});
    expect(wrapper.state().dialogSettings.period).toBe(888);

    freqUnit.simulate('change', {target: {value: 'days'}});
    expect(wrapper.state().dialogSettings.unit).toBe('days');

    var okButton = wrapper.find('.test-ok-button');
    expect(okButton).toHaveLength(1);
    okButton.first().simulate('click');

    // Dialog should be closed, link should now say "auto"
    expect(wrapper).toMatchSnapshot();
    expect(api.setWfModuleUpdateSettings.mock.calls.length).toBe(1);

    // '.toBe()' requires strict equality, and will fail
    expect(wrapper.state().liveSettings).toEqual({
      manual: false,
      period: 888,
      unit: 'days'
    });

    done();
  });

  it('Does not save changed settings when user hits Cancel', (done) => {

    modalLink.first().simulate('click');

    autoButton.first().simulate('click');
    freqNum.simulate('change', {target: {value: 888}});
    freqUnit.simulate('change', {target: {value: 'days'}});

    var cancelButton = wrapper.find('.test-cancel-button');
    expect(cancelButton).toHaveLength(1);
    cancelButton.first().simulate('click');

    // Dialog should be closed, link should now say "manual"
    expect(wrapper).toMatchSnapshot();
    // check that API was not called (was called once already in previous test)
    expect(api.setWfModuleUpdateSettings.mock.calls.length).toBe(1);

    // check that Live settings have NOT changed
    // '.toBe()' requires strict equality, and will fail
    expect(wrapper.state().liveSettings).toEqual({
      manual: true,
      period: 5,
      unit: 'minutes'
    });
    done();
  });

  it('Does not open modal when in read-only mode', (done) => {
    var readOnlyWrapper = shallow(
      <UpdateFrequencySelect
        isReadOnly={true}
        updateSettings={updateSettings}
        wfModuleId={212}
        api={api}
    />);
    expect(readOnlyWrapper.state().modalOpen).toBe(false);
    modalLink.first().simulate('click');
    expect(readOnlyWrapper.state().modalOpen).toBe(false);
    done();
  });

});
