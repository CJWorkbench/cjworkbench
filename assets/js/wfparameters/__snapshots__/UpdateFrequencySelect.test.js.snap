// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdateFrequencySelect Does not save changed settings when user hits Cancel 1`] = `
<div
  className="frequency-item"
>
  <div>
    <span
      className="content-3 t-d-gray"
    >
      Update 
    </span>
    <span
      className="content-3 ml-1 t-f-blue test-modal-button"
      onClick={[Function]}
    >
      manual
    </span>
  </div>
  <div
    className="content-4 t-m-gray"
  >
    Checked 
    2d ago
  </div>
  <Modal
    autoFocus={true}
    backdrop={true}
    backdropTransitionTimeout={150}
    className="modal-dialog"
    fade={true}
    isOpen={false}
    keyboard={true}
    modalTransitionTimeout={300}
    toggle={[Function]}
    zIndex={1050}
  >
    <ModalHeader
      className="dialog-header"
      tag="h4"
      toggle={[Function]}
      wrapTag="div"
    >
      <span
        className="title-4 t-d-gray"
      >
        SYNC SETTINGS
      </span>
    </ModalHeader>
    <ModalBody
      className="dialog-body"
      tag="div"
    >
      <FormGroup
        tag="div"
      >
        <div
          className="row"
        >
          <div
            className="col-sm-3"
          >
            <div
              className="action-button auto-button button-gray"
              onClick={[Function]}
            >
              On
            </div>
          </div>
          <div
            className="col-sm-9"
          >
            <div
              className="info-2"
            >
              Automatically save the current version of the workflow and update it with the newest data.
            </div>
            <Label
              className="content-3 t-d-gray mt-4 mb-2"
              for="updateFreq"
              tag="label"
            >
              Check for update every
            </Label>
            <div
              className="update-freq-settings update-freq-test-class mb-5"
            >
              <Input
                className="number-field t-d-gray content-2"
                id="updateFreqNum"
                max="500"
                min="1"
                name="updateFreq"
                onChange={[Function]}
                tag="p"
                type="number"
                value={5}
              />
              <Input
                className="ml-3 input-dropdown"
                id="updateFreqUnit"
                name="updateFreq"
                onChange={[Function]}
                tag="p"
                type="select"
                value="minutes"
              >
                <option>
                  minutes
                </option>
                <option>
                  hours
                </option>
                <option
                  id="days-option"
                >
                  days
                </option>
                <option>
                  weeks
                </option>
              </Input>
            </div>
          </div>
        </div>
        <div
          className="row d-flex align-items-center"
        >
          <div
            className="col-sm-3"
          >
            <div
              className="action-button manual-button button-blue--fill"
              onClick={[Function]}
            >
              Off
            </div>
          </div>
          <div
            className="col-sm-9"
          >
            <div
              className="info-2"
            >
              Check for new data manually.
            </div>
          </div>
        </div>
      </FormGroup>
    </ModalBody>
    <ModalFooter
      className="dialog-footer"
      tag="div"
    >
      <div
        className="action-button button-gray test-cancel-button mr-4"
        onClick={[Function]}
      >
        Cancel
      </div>
      <div
        className="action-button button-blue test-ok-button"
        onClick={[Function]}
      >
        Apply
      </div>
    </ModalFooter>
  </Modal>
</div>
`;

exports[`UpdateFrequencySelect Renders correctly when in Private mode and changes are confirmed with OK 1`] = `
<div
  className="frequency-item"
>
  <div>
    <span
      className="content-3 t-d-gray"
    >
      Update 
    </span>
    <span
      className="content-3 ml-1 t-f-blue test-modal-button"
      onClick={[Function]}
    >
      manual
    </span>
  </div>
  <div
    className="content-4 t-m-gray"
  >
    Checked 
    2d ago
  </div>
  <Modal
    autoFocus={true}
    backdrop={true}
    backdropTransitionTimeout={150}
    className="modal-dialog"
    fade={true}
    isOpen={false}
    keyboard={true}
    modalTransitionTimeout={300}
    toggle={[Function]}
    zIndex={1050}
  >
    <ModalHeader
      className="dialog-header"
      tag="h4"
      toggle={[Function]}
      wrapTag="div"
    >
      <span
        className="title-4 t-d-gray"
      >
        SYNC SETTINGS
      </span>
    </ModalHeader>
    <ModalBody
      className="dialog-body"
      tag="div"
    >
      <FormGroup
        tag="div"
      >
        <div
          className="row"
        >
          <div
            className="col-sm-3"
          >
            <div
              className="action-button auto-button button-gray"
              onClick={[Function]}
            >
              On
            </div>
          </div>
          <div
            className="col-sm-9"
          >
            <div
              className="info-2"
            >
              Automatically save the current version of the workflow and update it with the newest data.
            </div>
            <Label
              className="content-3 t-d-gray mt-4 mb-2"
              for="updateFreq"
              tag="label"
            >
              Check for update every
            </Label>
            <div
              className="update-freq-settings update-freq-test-class mb-5"
            >
              <Input
                className="number-field t-d-gray content-2"
                id="updateFreqNum"
                max="500"
                min="1"
                name="updateFreq"
                onChange={[Function]}
                tag="p"
                type="number"
                value={5}
              />
              <Input
                className="ml-3 input-dropdown"
                id="updateFreqUnit"
                name="updateFreq"
                onChange={[Function]}
                tag="p"
                type="select"
                value="minutes"
              >
                <option>
                  minutes
                </option>
                <option>
                  hours
                </option>
                <option
                  id="days-option"
                >
                  days
                </option>
                <option>
                  weeks
                </option>
              </Input>
            </div>
          </div>
        </div>
        <div
          className="row d-flex align-items-center"
        >
          <div
            className="col-sm-3"
          >
            <div
              className="action-button manual-button button-blue--fill"
              onClick={[Function]}
            >
              Off
            </div>
          </div>
          <div
            className="col-sm-9"
          >
            <div
              className="info-2"
            >
              Check for new data manually.
            </div>
          </div>
        </div>
      </FormGroup>
    </ModalBody>
    <ModalFooter
      className="dialog-footer"
      tag="div"
    >
      <div
        className="action-button button-gray test-cancel-button mr-4"
        onClick={[Function]}
      >
        Cancel
      </div>
      <div
        className="action-button button-blue test-ok-button"
        onClick={[Function]}
      >
        Apply
      </div>
    </ModalFooter>
  </Modal>
</div>
`;

exports[`UpdateFrequencySelect Renders correctly when in Private mode and changes are confirmed with OK 2`] = `
<div
  className="frequency-item"
>
  <div>
    <span
      className="content-3 t-d-gray"
    >
      Update 
    </span>
    <span
      className="content-3 ml-1 t-f-blue test-modal-button"
      onClick={[Function]}
    >
      manual
    </span>
  </div>
  <div
    className="content-4 t-m-gray"
  >
    Checked 
    2d ago
  </div>
  <Modal
    autoFocus={true}
    backdrop={true}
    backdropTransitionTimeout={150}
    className="modal-dialog"
    fade={true}
    isOpen={true}
    keyboard={true}
    modalTransitionTimeout={300}
    toggle={[Function]}
    zIndex={1050}
  >
    <ModalHeader
      className="dialog-header"
      tag="h4"
      toggle={[Function]}
      wrapTag="div"
    >
      <span
        className="title-4 t-d-gray"
      >
        SYNC SETTINGS
      </span>
    </ModalHeader>
    <ModalBody
      className="dialog-body"
      tag="div"
    >
      <FormGroup
        tag="div"
      >
        <div
          className="row"
        >
          <div
            className="col-sm-3"
          >
            <div
              className="action-button auto-button button-gray"
              onClick={[Function]}
            >
              On
            </div>
          </div>
          <div
            className="col-sm-9"
          >
            <div
              className="info-2"
            >
              Automatically save the current version of the workflow and update it with the newest data.
            </div>
            <Label
              className="content-3 t-d-gray mt-4 mb-2"
              for="updateFreq"
              tag="label"
            >
              Check for update every
            </Label>
            <div
              className="update-freq-settings update-freq-test-class mb-5"
            >
              <Input
                className="number-field t-d-gray content-2"
                id="updateFreqNum"
                max="500"
                min="1"
                name="updateFreq"
                onChange={[Function]}
                tag="p"
                type="number"
                value={5}
              />
              <Input
                className="ml-3 input-dropdown"
                id="updateFreqUnit"
                name="updateFreq"
                onChange={[Function]}
                tag="p"
                type="select"
                value="minutes"
              >
                <option>
                  minutes
                </option>
                <option>
                  hours
                </option>
                <option
                  id="days-option"
                >
                  days
                </option>
                <option>
                  weeks
                </option>
              </Input>
            </div>
          </div>
        </div>
        <div
          className="row d-flex align-items-center"
        >
          <div
            className="col-sm-3"
          >
            <div
              className="action-button manual-button button-blue--fill"
              onClick={[Function]}
            >
              Off
            </div>
          </div>
          <div
            className="col-sm-9"
          >
            <div
              className="info-2"
            >
              Check for new data manually.
            </div>
          </div>
        </div>
      </FormGroup>
    </ModalBody>
    <ModalFooter
      className="dialog-footer"
      tag="div"
    >
      <div
        className="action-button button-gray test-cancel-button mr-4"
        onClick={[Function]}
      >
        Cancel
      </div>
      <div
        className="action-button button-blue test-ok-button"
        onClick={[Function]}
      >
        Apply
      </div>
    </ModalFooter>
  </Modal>
</div>
`;

exports[`UpdateFrequencySelect Renders correctly when in Private mode and changes are confirmed with OK 3`] = `
<div
  className="frequency-item"
>
  <div>
    <span
      className="content-3 t-d-gray"
    >
      Update 
    </span>
    <span
      className="content-3 ml-1 t-f-blue test-modal-button"
      onClick={[Function]}
    >
      auto
    </span>
  </div>
  <div
    className="content-4 t-m-gray"
  >
    Checked 
    2d ago
  </div>
  <Modal
    autoFocus={true}
    backdrop={true}
    backdropTransitionTimeout={150}
    className="modal-dialog"
    fade={true}
    isOpen={false}
    keyboard={true}
    modalTransitionTimeout={300}
    toggle={[Function]}
    zIndex={1050}
  >
    <ModalHeader
      className="dialog-header"
      tag="h4"
      toggle={[Function]}
      wrapTag="div"
    >
      <span
        className="title-4 t-d-gray"
      >
        SYNC SETTINGS
      </span>
    </ModalHeader>
    <ModalBody
      className="dialog-body"
      tag="div"
    >
      <FormGroup
        tag="div"
      >
        <div
          className="row"
        >
          <div
            className="col-sm-3"
          >
            <div
              className="action-button auto-button button-blue--fill"
              onClick={[Function]}
            >
              On
            </div>
          </div>
          <div
            className="col-sm-9"
          >
            <div
              className="info-2"
            >
              Automatically save the current version of the workflow and update it with the newest data.
            </div>
            <Label
              className="content-3 t-d-gray mt-4 mb-2"
              for="updateFreq"
              tag="label"
            >
              Check for update every
            </Label>
            <div
              className="update-freq-settings update-freq-test-class mb-5"
            >
              <Input
                className="number-field t-d-gray content-2"
                id="updateFreqNum"
                max="500"
                min="1"
                name="updateFreq"
                onChange={[Function]}
                tag="p"
                type="number"
                value={888}
              />
              <Input
                className="ml-3 input-dropdown"
                id="updateFreqUnit"
                name="updateFreq"
                onChange={[Function]}
                tag="p"
                type="select"
                value="days"
              >
                <option>
                  minutes
                </option>
                <option>
                  hours
                </option>
                <option
                  id="days-option"
                >
                  days
                </option>
                <option>
                  weeks
                </option>
              </Input>
            </div>
          </div>
        </div>
        <div
          className="row d-flex align-items-center"
        >
          <div
            className="col-sm-3"
          >
            <div
              className="action-button manual-button button-gray "
              onClick={[Function]}
            >
              Off
            </div>
          </div>
          <div
            className="col-sm-9"
          >
            <div
              className="info-2"
            >
              Check for new data manually.
            </div>
          </div>
        </div>
      </FormGroup>
    </ModalBody>
    <ModalFooter
      className="dialog-footer"
      tag="div"
    >
      <div
        className="action-button button-gray test-cancel-button mr-4"
        onClick={[Function]}
      >
        Cancel
      </div>
      <div
        className="action-button button-blue test-ok-button"
        onClick={[Function]}
      >
        Apply
      </div>
    </ModalFooter>
  </Modal>
</div>
`;
